packages/server/modules/workspaces/tests/integration/projects.graph.spec.ts

```diff
@@ -0,0 +1,182 @@
+import { AllScopes } from '@/modules/core/helpers/mainConstants'
+import {
+  BasicTestWorkspace,
+  createTestWorkspace
+} from '@/modules/workspaces/tests/helpers/creation'
+import {
+  BasicTestUser,
+  createAuthTokenForUser,
+  createTestUser
+} from '@/test/authHelper'
+import {
+  CreateWorkspaceProjectDocument,
+  GetWorkspaceProjectsDocument
+} from '@/test/graphql/generated/graphql'
+import {
+  createTestContext,
+  testApolloServer,
+  TestApolloServer
+} from '@/test/graphqlHelper'
+import { beforeEachContext } from '@/test/hooks'
+import { Roles } from '@speckle/shared'
+import { expect } from 'chai'
+import cryptoRandomString from 'crypto-random-string'
+
+describe('Workspace project GQL CRUD', () => {
+  let apollo: TestApolloServer
+
+  const testUser: BasicTestUser = {
+    id: '',
+    name: 'John Speckle',
+    email: 'john-speckle-workspace-project-admin@example.org',
+    role: Roles.Server.Admin
+  }
+
+  const testNonWorkspaceMemberUser: BasicTestUser = {
+    id: '',
+    name: 'John Nobody',
+    email: 'john-nobody@example.org',
+    role: Roles.Server.User
+  }
+
+  before(async () => {
+    await beforeEachContext()
+    await Promise.all(
+      [testUser, testNonWorkspaceMemberUser].map((user) => createTestUser(user))
```

Comment:
No need to change this, but there's also `createTestUsers` (plural) that would remove the need to do this Promise.all & mapping bit

Reply:
I need to learn to read, done